### ISSUE TITLE
The `amp-social-share` component is not properly including data-param attributes in the shared URL when using type='twitter'.

### ISSUE THREAD

**Issue by commadelimited** [ampproject/amphtml#15354]
## What's the issue?

The `amp-social-share` component is not properly including data-param attributes in the shared URL when using type='twitter'.

## How do we reproduce the issue?

1. Visit [this AMP page on Eventbrite.com](https://www.eventbrite.com/e/snakehips-poolside-day-party-reschedule-527-tickets-43459998024/amp).
2. Inspect the Twitter sharing button and observe the attribute `data-param-attribute` is included on the component
3. Click the Twitter sharing button and observe that no query params are included on the shared URL.

## What browsers are affected?
* Chrome 65.0.3325.181
* Firefox 59.0.2
* Safari 10.1.2

## Which AMP version is affected?
```
amp-version="1525461683159"
<script async src="https://cdn.ampproject.org/v0.js"></script>
<script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"></script>
```
## Is this a new issue?

Unsure when this started happening. This issue was first reported to us in September of 2017, but is still happening today.

## Additional details

I can reproduce this issue in my development environment. I've tried including additional params and experience the same issue. I feel like the documentation here is _very_ clear, and it appears that we're following the docs to the letter. So I can only assume this is some odd bug we're experiencing.

Here's the code we're using to output a sharing button:
```

${ helpers.social_share_component('twitter', params=[{ 'name': 'foo', 'value': 'bar'}]) }

<%def name="social_share_component(type, params=None)">
    <amp-social-share
        width="40"
        height="40"
        class="amp-listing-event-social-share__icon amp-l-mar-right-1"
        type="${ type }"
        % if params:
            % for param in params:
                data-param-${ param.get('name') }="${ param.get('value') }"
            % endfor
        % endif
    >
    </amp-social-share>
</%def>
```
Which results in the following output
```
<amp-social-share
    width="40"
    height="40"
    class="amp-listing-event-social-share__icon amp-l-mar-right-1 i-amphtml-element i-amphtml-layout-fixed i-amphtml-layout-size-defined amp-social-share-twitter i-amphtml-layout"
    type="twitter"
    data-param-attribute="amplisting"
    data-param-foo="bar"
    style="width: 40px; height: 40px;"
    role="button"
    tabindex="0"
>
</amp-social-share>
```
Clicking the Twitter icon results in the following behaviour:
![screen shot 2018-05-16 at 11 30 53 am](https://user-images.githubusercontent.com/563460/40130450-b0cae44c-58fc-11e8-93d7-3c9266b12773.png)

**Comment by aghassemi** [comment]
@commadelimited Thanks for the report, following the repro stepson Chrome, I still see the params getting added to the URL. 

```
https://twitter.com/intent/tweet?text=Eventbrite%20%7C%20SNAKEHIPS%20-%20Poolside%20Day%20Party%20%2F%20Reschedule%205.27&url=https%3A%2F%2Fwww.eventbrite.com%2Fe%2Fsnakehips-poolside-day-party-reschedule-527-tickets-43459998024&attribute=amplisting
```

Note the `attribute=amplisting` added at the end which is comes from `data-param-attribute="amplisting"`.

What's the URL you are seeing? (the end of the Url is cutoff in the screenshot above)

**Comment by commadelimited** [comment]
Interesting! Let me follow STR again. (URL isn't cut off in the screenshot, that's the entire thing btw. I even enlarged the window to make sure it got everything.

Safari (copying directly from the Twitter share box):
```
Eventbrite | SNAKEHIPS - Poolside Day Party / Reschedule 5.27 https://www.eventbrite.com/e/snakehips-poolside-day-party-reschedule-527-tickets-43459998024
```
Chrome (copying directly from the Twitter share box):
```
Eventbrite | SNAKEHIPS - Poolside Day Party / Reschedule 5.27 https://www.eventbrite.com/e/snakehips-poolside-day-party-reschedule-527-tickets-43459998024
```
Firefox (copying directly from the Twitter share box):
```
Eventbrite | SNAKEHIPS - Poolside Day Party / Reschedule 5.27 https://www.eventbrite.com/e/snakehips-poolside-day-party-reschedule-527-tickets-43459998024
```

**Comment by commadelimited** [comment]
Taking another look at this, would be it be easiest to simply include the complete URL we'd like to share in the data-url attribute? This URL could include all required query params. Thoughts

**Comment by aghassemi** [comment]
@commadelimited Correct, the `data-param-*` gets added to the share endpoint not the Url you are sharing. It is intended for extra configurations that some share endpoints accept (e.g. pinterest accepts a `media` param) or for changing existing params such as title and url.

You can do something like `<amp-social-share type="twitter" data-param-url="CANONICAL_URL?foo=bar"></amp-social-share>` if you like to change the share Url itself.

**Comment by commadelimited** [comment]
Ahhh. Once you explain it, that makes sense. I'd really suggest updating the documentation to make that more clear though. Given the description on the [amp-social-share documentation for data-param*](https://www.ampproject.org/docs/reference/components/amp-social-share#data-param-*) it's not an unreasonable expectation that the URL in the share box itself would include the URL params.
![screen shot 2018-05-17 at 1 31 02 pm](https://user-images.githubusercontent.com/563460/40196638-bd7f416c-59d6-11e8-86a9-3489a1ac912f.png)

Thanks for your time though @aghassemi, appreciated.
